scan-translations.ts

```diff
@@ -0,0 +1,383 @@
+
+    for (const pattern of TRANSLATION_PATTERNS) {
+        let match;
+        // Reset lastIndex for global regex
+        pattern.lastIndex = 0;
+
+        while ((match = pattern.exec(content)) !== null) {
+            const key = match[1];
+            // Skip dynamic keys (containing variables like ${}, {{}} etc.)
+            if (!key.includes("${") && !key.includes("{{") && !key.includes("}")) {
+                keys.push(key);
+            }
+        }
+    }
+
+    return keys;
+}
+
+/**
+ * Scan source files for translation keys
+ */
+async function scanSourceFiles(): Promise<Set<string>> {
+    console.log("🔍 Scanning source files for translation keys...\n");
+
+    const usedKeys = new Set<string>();
+
+    // Find all TypeScript and TypeScript React files
+    const files = await glob("**/*.{ts,tsx}", {
+        cwd: WEB_APP_DIR,
+        ignore: EXCLUDE_DIRS,
+        absolute: true,
+    });
+
+    console.log(`Found ${files.length} files to scan\n`);
+
+    for (const file of files) {
+        try {
+            const content = fs.readFileSync(file, "utf-8");
+            const keys = extractKeysFromContent(content);
+
+            keys.forEach(key => usedKeys.add(key));
+        } catch (error) {
+            console.error(`❌ Error: Could not read file ${file}: ${error}`);
+        }
+    }
+
+    console.log(`✅ Found ${usedKeys.size} unique translation keys in source files\n`);
+
+    return usedKeys;
+}
+
+/**
+ * Get all locale files in the locales directory
+ */
+function getLocaleFiles(): string[] {
+    const files = fs.readdirSync(LOCALES_DIR);
+    return files
+        .filter(file => file.endsWith('.json'))
+        .map(file => file.replace('.json', ''));
+}
+
+/**
+ * Load translation keys from a specific locale file
+ */
+function loadKeysFromLocale(locale: string): Set<string> {
+    const localePath = path.join(LOCALES_DIR, `${locale}.json`);
+    const translationKeys = new Set<string>();
+
+    if (!fs.existsSync(localePath)) {
+        throw new Error(`❌ Locale file not found: ${localePath}`);
+    }
+
+    try {
+        const content = fs.readFileSync(localePath, "utf-8");
+        const translations = JSON.parse(content);
+        const keys = flattenKeys(translations);
+
+        keys.forEach(key => translationKeys.add(key));
+
+        return translationKeys;
+    } catch (error) {
+        throw new Error(`❌ Failed to parse ${localePath}: ${error}`);
+    }
+}
+
+/**
+ * Load translation keys from all locale files
+ */
+function loadAllTranslationKeys(): Map<string, Set<string>> {
+    console.log("📚 Loading translation keys from locale files...\n");
+
+    const allLocales = getLocaleFiles();
+    const translationsByLocale = new Map<string, Set<string>>();
+
+    for (const locale of allLocales) {
+        const keys = loadKeysFromLocale(locale);
+        translationsByLocale.set(locale, keys);
+        console.log(`   • ${locale}.json: ${keys.size} keys`);
+    }
+
+    console.log();
```

Comment:
no need of a separate console.log here to print a new line, please add a `\n` character at the end of the previous log if this is the use case for the empty log, if not pls remove it 🙏